grails-app/assets/javascripts/search.js

```diff
@@ -256,7 +256,7 @@ $(document).ready(function() {
         var index = i; // keep a copy
         var queryContextParam = (BC_CONF.queryContext) ? "&qc=" + BC_CONF.queryContext : "";
         var jsonUri = BC_CONF.biocacheServiceUrl + "/occurrences/search.json?q=lsid:" + lsid + "&" + BC_CONF.facetQueries +
-            "&facets=raw_taxon_name&pageSize=0&flimit=" + maxFacets + queryContextParam + "&callback=?";
+            "&facets=raw_taxon_name&pageSize=0&flimit=" + maxFacets + queryContextParam;  // + "&callback=?";
```

Comment:
Is the lack of a 'callback' parameter automatically detected to switch from JSONP to JSON in this case?

Reply:
In this case, kind of... the problem wasn't JSONP but the service was still appending the callback to the request, regardless of whether JSONP was being requested. This resulted in invalid JSON. Removing the callback params fixed the invalid JSON.